### PR TITLE
Snippets generation feature

### PR DESCRIPTION
**Pull request by YuryShkoda**
## Intent

- Implement `sasjs` command to generate VS Code snippets from SAS Macros.

## Implementation

- Added `sasjs snippets` command.
- Covered `sasjs snippets` command with unit tests.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
- [x] Unit tests coverage has been increased and a new threshold is set.
- [x] All CI checks are green.
- [x] Development comments have been added or updated.
- [x] Development documentation coverage has been increased and a new threshold is set.
- [x] Reviewer is assigned.

### Reviewer checks

- [ ] Any new code is documented.

### CODE REVIEW

File: src/commands/snippets/snippetsCommand.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { CommandExample, ReturnCode } from '../../types/command'
+import { TargetCommand } from '../../types/command/targetCommand'
+
+const syntax = 'snippets'
+const usage = 'sasjs snippets'
+const description = `Generates VS Code snippets from SAS Macros.`
+const examples: CommandExample[] = [
+  {
+    command: 'sasjs snippets',
+    description: 'Generates VS Code snippets from SAS Macros.'
```
[the remark this answers is not in the document]
Comment (YuryShkoda) [reply src/commands/snippets/snippetsCommand.ts]: done

File: src/types/command/commandFactory.ts [hunk 2]
```diff
@@ -48,7 +49,8 @@ export const commandFactory = new Map<string, Function>([
   ['servicepack', (args: string[]) => new ServicePackCommand(args)],
   ['version', (args: string[]) => new VersionCommand(args)],
   ['test', (args: string[]) => new TestCommand(args)],
-  ['web', (args: string[]) => new WebCommand(args)]
+  ['web', (args: string[]) => new WebCommand(args)],
+  ['snippets', (args: string[]) => new SnippetsCommand(args)]
```
Comment (allanbowe) [inline src/types/command/commandFactory.ts:53]: would be nice if this retained the alphabetical order

File: src/commands/snippets/snippets.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  // return file path with generated VS Code snippets
+  return Promise.resolve(snippetsFilePath)
+}
+
+/**
+ * Creates a VS Code snippet from SAS macro.
+ * @param file file path of SAS macro.
+ * @returns promise that resolves with VS Code snippet.
+ */
+const createMacro = async (file: string): Promise<Snippet> => {
```
Comment (sabhas) [inline src/commands/snippets/snippets.ts:87]: I think the function name should be updated. Add something about the snippet.

**Review by sabhas — CHANGES_REQUESTED** [review]

File: src/commands/snippets/snippets.ts (hunk 3 above)
Comment (YuryShkoda) [reply src/commands/snippets/snippets.ts]: makes sense, fixed

**Review by allanbowe — APPROVED** [review]

**Review by sabhas — APPROVED** [review]

**Pull request merged**